src/Raven.Server/Web/System/AdminDumpHandler.cs

```diff
@@ -0,0 +1,163 @@
+
+                    using (var stream = ResponseBodyStream())
+                        await file.CopyToAsync(stream);
+                }
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.Synchronized)]
+        private static void Execute(string args, int processId, string output)
+        {
+            var ravenDebugExec = Path.Combine(AppContext.BaseDirectory,
+                PlatformDetails.RunningOnPosix ? "Raven.Debug" : "Raven.Debug.exe"
+            );
+
+            if (File.Exists(ravenDebugExec) == false)
+                throw new FileNotFoundException($"Could not find debugger tool at '{ravenDebugExec}'");
+
+            var sb = new StringBuilder($"{args} --pid {processId} --output {CommandLineArgumentEscaper.EscapeSingleArg(output)}");
+
+            var startup = new ProcessStartInfo
+            {
+                Arguments = sb.ToString(),
+                FileName = ravenDebugExec,
+                WindowStyle = ProcessWindowStyle.Normal,
+                RedirectStandardError = true,
+                RedirectStandardOutput = true,
+                RedirectStandardInput = true,
+                UseShellExecute = false
+            };
+
+            if (PlatformDetails.RunningOnPosix == false)
+                startup.LoadUserProfile = false;
+
+            var process = new Process
+            {
+                StartInfo = startup,
+                EnableRaisingEvents = true
+            };
+
+            sb.Clear();
+            process.ErrorDataReceived += (sender, args) => sb.Append(args.Data);
+
+            process.Start();
+
+            process.BeginErrorReadLine();
+            process.BeginOutputReadLine();
+
+            process.WaitForExit();
+
+            if (process.ExitCode != 0)
+                throw new InvalidOperationException($"Could not read stack traces, exit code: {process.ExitCode}, error: {sb}");
+        }
+
+        private string GetFileName(string extension)
+        {
+            var nodeTag = ServerStore.NodeTag == RachisConsensus.InitialTag
+                ? "Unknown"
+                : ServerStore.NodeTag;
+
+            var platform = PlatformDetails.RunningOnPosix
+                ? ".linux"
+                : ".win";
+
+            if (PlatformDetails.Is32Bits)
+                platform += "-x86";
+            else
+                platform += "-x64";
+
+            return $"RavenDB_{RavenVersionAttribute.Instance.BuildVersion}_{nodeTag}_{DateTime.Now.ToString("ddMMyyyy_HHmmss")}{platform}{extension}";
+        }
+
+        private class DeleteFile : IDisposable
+        {
+            private readonly string _path;
+
+            public DeleteFile(string path)
+            {
+                if (path == null)
+                    throw new ArgumentNullException(nameof(path));
+
+                _path = path;
+            }
+
+            public void Dispose()
+            {
+                if (File.Exists(_path))
```

Comment:
Can skip the exists, DeleteFile handles it.

Reply:
Actually it doesn't